test/NuGet.Server.Tests/ServerPackageStoreTest.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System.IO;
+using System.Text;
+using Moq;
+using NuGet.Server.Infrastructure;
+using Xunit;
+
+namespace NuGet.Server.Tests
+{
+    public class ServerPackageStoreTest
+    {
+        [Theory]
+        [InlineData("[")]
+        [InlineData("]")]
+        [InlineData("{")]
+        [InlineData("}")]
+        [InlineData("[{")]
+        [InlineData("[{}")]
+        [InlineData("{}")]
+        [InlineData("[{\"foo\": \"bar\"}]")]
+        public void Constructor_IgnoresAndDeletesInvalidCacheFile(string content)
```

Comment:
A test to verify that the file isn't deleted for valid json would be helpful. It seems like the different inputs here are just testing json.net not nuget.server.